contrib/hooks/clang-format-pre-commit.sh

```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+# Copyright (c) 2021 The Orbit Authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+# A pre-commit hook that applies clang-format-9 to all the files touched by this
+# commit.
+# Note, it will format and add the complete file, and thus does not work with
+# hunks.
+
+echo "Formatting changes now..."
+while read line; do
+  echo "Formatting ${line}"
+  clang-format-9 -i ${line}
+  git add ${line}
+done <<< $(git diff-index --cached --name-only HEAD | grep -iE '\.(h|cpp|cc|proto)')
```

Comment:
I am not sure if this one has the same behavior, but if I amend commit to amend only a commit message (so there are no changes) it displays a error. Maybe add a check that the list of modified files is not empty?

Reply:
Should be better now.